### ISSUE TITLE
Raw data vs resps_data()

### ISSUE THREAD

**Issue by jonthegeek** [r-lib/httr2#621]
When fetching images or other raw content from an API with multiple requests, `resps_data()` seems to work, but it merges the raw results into a single raw vector. This technically makes sense with what the function say it will do, but doesn't feel like what the user would expect. I ran into this fetching images, and briefly couldn't figure out where the images after the first one went (magick ignored the extra raw data for the most part, so the first image "absorbed" the rest).

I don't know if this should be covered in documentation, if it makes sense to have a separate function for extracting raw vectors from responses into a list, or if it's good enough as-is and should be left to the user to figure out, but I wanted to raise the flag and see what you thought.

``` r
# Fetch images from an API.
req <- list(httr2::request("https://picsum.photos/100"))
reqs <- rep(req, 3)
resps <- httr2::req_perform_sequential(reqs)
#> Iterating ■■■■■■■■■■■ 33% | ETA: 3s

# resps_data seems to work to extract data, but merges the images.
results_naive <- httr2::resps_data(resps, httr2::resp_body_raw)
length(results_naive)
#> [1] 7892
class(results_naive)
#> [1] "raw"
# Images 2 and 3 are "inside" image 1
magick::image_read(results_naive)
```

<img src="https://i.imgur.com/FqtqdiL.png" width="100" />

``` r

# Listifying the raw results works.
resp_parser <- function(resp) {
  list(httr2::resp_body_raw(resp))
}
results_split <- httr2::resps_data(resps, resp_parser)
length(results_split)
#> [1] 3
class(results_split)
#> [1] "list"
class(results_split[[1]])
#> [1] "raw"
magick::image_read(results_split[[1]])
```

<img src="https://i.imgur.com/cwKbAlI.png" width="100" />

``` r
magick::image_read(results_split[[2]])
```

<img src="https://i.imgur.com/4hXk0rU.png" width="100" />

``` r
magick::image_read(results_split[[3]])
```

<img src="https://i.imgur.com/pW4MWJI.png" width="100" />

<sup>Created on 2024-12-26 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by hadley** [comment]
This feels like maybe worth a note in the docs somewhere, since I don't think `resps_data(resps, (resp) list(resp_body_raw(res)))` needs its own helper.

**Comment by jonthegeek** [comment]
Yeah, that makes sense. Mostly it was just surprising since I was seeing what came back and expected it to error out (or give me a bad image) until I really thought about what was happening to the raw vector.
